hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java

```diff
@@ -528,6 +557,15 @@ private ContainerCommandResponseProto dispatchCommand(
   private CompletableFuture<Message> writeStateMachineData(
       ContainerCommandRequestProto requestProto, long entryIndex, long term,
       long startTime) {
+    if (writeChunkFutureMap.containsKey(entryIndex)) {
+      // generally state machine will wait forever, for precaution, a check is added if retry happens.
+      return writeChunkFutureMap.get(entryIndex).getRaftFuture();
+    }
+    try {
+      validateLongRunningWrite(entryIndex);
+    } catch (IOException e) {
```

Comment:
Do we want to have another exception handling here? Even if this gets submitted to the threadpool it will fail since we have marked the statemachine as unhealthy. Do we want this optmization to fail fast here? Eventually the pipeline will close anyhow.

Reply:
We can call the check checkStateMachineUnhealthy function instead

Reply:
from exception perspective, its more clear here why exception has happened, later for further write(), it will throw unhealty exception by checkStateMachineUnhealthy. So difference in exception will provide clarity.
IMO, no need change this.